monai/metrics/metric.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Any, Optional
+
+import torch
+
+from monai.config import TensorList
+
+
+class Metric(ABC):
```

Comment:
@Nic-Ma seems like you already support list of tensors here. Maybe, I'm missing the context here. Could you please explain how this will be working with ignite metrics ? Thanks

Reply:
Actually, we have 2 levels of metrics in MONAI:
1. PyTorch metrics, this `Metric` base class is for these metrics.
2. ignite metrics, we have many ignite metrics under the handlers/ folder, they are wrapper of 1, but inherit ignite Metric.

This PR added support for `list of Tensor` in all the MONAI metrics, but if your ignite metrics can't support it, we can't use ignite metrics directly.

Thanks.